pymc3/distributions/multivariate.py

```diff
@@ -514,7 +493,11 @@ def random(self, point=None, size=None):
         array
         """
         a = draw_values([self.a], point=point, size=size)[0]
-        samples = generate_samples(self._random, a=a, dist_shape=self.shape, size=size)
+        output_shape = to_tuple(size) + a.shape
+        a = broadcast_dist_samples_to(to_shape=output_shape, samples=[a], size=size)[0]
+        samples = stats.gamma.rvs(a=a, size=output_shape)
+        samples = samples / samples.sum(-1, keepdims=True)
+        samples = samples.reshape(output_shape)
```

Comment:
Why do we need reshape here?

Reply:
Shapes like (1, x, y) were getting simplified into simply (x, y). Reshaping for handling those issues.

Reply:
Can you share an example on this? Because if this is the case, we need to modify all PyMC3 multivariate distributions' `random` method to avoid dropping the first dimension.

Reply:
Any case with `size` parameter as `1` will behave that way. A few such cases are in the newly added tests.

Reply:
I checked out this PR locally and commented out reshape line. All the tests passed through 
```bash
pytest -v pymc3/tests/test_distributions_random.py::TestDirichlet::test_dirichlet_random_shape
```
Can you check if this is the same behaviour at your side?

Reply:
Ah yes, must be a residual from a previous code combination. I'll remove it shortly